pandas/core/strings.py

```diff
@@ -1859,7 +1827,11 @@ def _validate(data):
 
             # see src/inference.pyx which can contain string values
             allowed_types = ('string', 'unicode', 'mixed', 'mixed-integer')
-            if data.inferred_type not in allowed_types:
+            if is_categorical_dtype(data.dtype):
```

Comment:
you should be able to 
``Index(data).inferred_type`` here (as if its not an Index then it will be for inference, otherwise it will just work)

Reply:
after ``isinstance(data, Index)``, data is already an Index. The previous ``if``-check was failing for categorical data (which isn't in the list of allowed types). But maybe I misunderstand something?

Reply:
I see, ok then